mempool/clist_mempool_test.go

```diff
@@ -170,23 +170,37 @@ func TestMempoolFilters(t *testing.T) {
 		{10, PreCheckAminoMaxBytes(22), PostCheckMaxGas(0), 0},
 	}
 	for tcIndex, tt := range tests {
-		mempool.Update(1, emptyTxArr, tt.preFilter, tt.postFilter)
+		mempool.Update(1, emptyTxArr, abciResponses(len(emptyTxArr), abci.CodeTypeOK), tt.preFilter, tt.postFilter)
 		checkTxs(t, mempool, tt.numTxsToCreate, UnknownPeerID)
 		require.Equal(t, tt.expectedNumTxs, mempool.Size(), "mempool had the incorrect size, on test case %d", tcIndex)
 		mempool.Flush()
 	}
 }
 
-func TestMempoolUpdateAddsTxsToCache(t *testing.T) {
+func TestMempoolUpdate(t *testing.T) {
 	app := kvstore.NewKVStoreApplication()
 	cc := proxy.NewLocalClientCreator(app)
 	mempool, cleanup := newMempoolWithApp(cc)
 	defer cleanup()
-	mempool.Update(1, []types.Tx{[]byte{0x01}}, nil, nil)
+
+	// 1. Adds valid txs to the cache
+	mempool.Update(1, []types.Tx{[]byte{0x01}}, abciResponses(1, abci.CodeTypeOK), nil, nil)
 	err := mempool.CheckTx([]byte{0x01}, nil)
```

Comment:
Maybe stupid question: why is the order here: 1) Update 2) CheckTx while it is the other way around in the remaining parts of the test?

Reply:
if we execute mempool.CheckTx([]byte{0x01}, nil) before Update, it will be added to the cache and mempool
 - Update function will remove it from the mempool (2. test case), but leave it in the cache (noop)

now because we want to actually test that Update adds tx to cache, we can't add tx (call CheckTx) before

Reply:
https://github.com/tendermint/tendermint/pull/3625/commits/87db96b96d9f838e16d6843a4eb32910fcee0d7a

Reply:
Thanks for the explanation 💚